website/src/documentation/transforms/python/element-wise/filter.md

```diff
@@ -18,25 +18,168 @@ See the License for the specific language governing permissions and
+        width="20px" height="20px" alt="View on GitHub" />
+      View on GitHub
+    </a>
+  </td>
+</table>
+<br>
+
+### Multiple Arguments
+
+Multiple function arguments can be passed as additional arguments to `Filter`.
+They will be passed as additional positional arguments or keyword arguments to the function.
+
+```py
+{% github_sample /apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/transforms/element_wise/filter.py tag:filter_multiple_arguments %}```
+
+<table>
+  <td>
+    <a class="button" target="_blank"
+        href="https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/transforms/element_wise/filter.py">
+      <img src="https://www.tensorflow.org/images/GitHub-Mark-32px.png"
+        width="20px" height="20px" alt="View on GitHub" />
+      View on GitHub
+    </a>
+  </td>
+</table>
+<br>
+
+### Side Inputs - Singleton
+
+If there is a single value in the `PCollection`, such as the average from another computation,
+passing the `PCollection` as a *singleton* will access that value.
+
+```py
+{% github_sample /apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/transforms/element_wise/filter.py tag:filter_side_inputs_singleton %}```
+
+<table>
+  <td>
+    <a class="button" target="_blank"
+        href="https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/transforms/element_wise/filter.py">
+      <img src="https://www.tensorflow.org/images/GitHub-Mark-32px.png"
+        width="20px" height="20px" alt="View on GitHub" />
+      View on GitHub
+    </a>
+  </td>
+</table>
+<br>
+
+### Side Inputs - Iterator
+
+If there are multiple values in the `PCollection`, it is recommended way to pass the `PCollection` as an *iterator*.
+This will access elements lazily as they are needed,
+so it is possible to iterate over very large `PCollection`s that don't fit into memory.
+
+```py
+{% github_sample /apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/transforms/element_wise/filter.py tag:filter_side_inputs_iter %}```
+
+<table>
+  <td>
+    <a class="button" target="_blank"
+        href="https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/transforms/element_wise/filter.py">
+      <img src="https://www.tensorflow.org/images/GitHub-Mark-32px.png"
+        width="20px" height="20px" alt="View on GitHub" />
+      View on GitHub
+    </a>
+  </td>
+</table>
+<br>
+
+> **Note**: It is also possible to pass the `PCollection` as a *list* with `beam.pvalue.AsList(pcollection)`,
+> but this will require all the elements to fit into memory.
+
+### Side Inputs - Dict
+
+If the `PCollection` consists of a small enough number of key-value pairs that can fit into memory,
+the `PCollection` can be passed as a *dictionary*.
+Note that all the elements of the `PCollection` must fit into memory for this.
+If the `PCollection` can be very large, use `beam.pvalue.AsIter(pcollection)` instead.
```

Comment:
`can be very large,` -> `won't fit in memory,` ? (to use the similar terminology as above.)

Reply:
Makes sense, changed